### PR TITLE
Crud tasks

### PR DESCRIPTION
**Pull request by carreraprogrammer**
- [x] Remove all hardcoded items from the tasks array.
- [x] Create a new JavaScript file for the new functionality.
- [x] Implement a function for adding a new task (add a new element to the array).
- [x] Implement a function for deleting a task (remove an element from the array).
- [x] Implement a function for editing task descriptions.
- [x] By default new tasks should have the property completed set to false and the property index set to the value of the new array length
- [x] Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique
- [x] All changes to the To Do List should be saved in local storage.

### CODE REVIEW

File: src/modules/task.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        })
+      }
+    
+      static deleteTask(trash) {
+        return trash.addEventListener('click', (e) => {
+            e.target.parentElement.remove();
+            LocalStorage.removeTask(e.target.parentElement.id);
+            DisplayList.showList();
+          });
+      }
```
Comment (ShadyShawkat) [inline src/modules/task.js:44]: - [ ] Please make sure that the 'id's are reset and ordered again for the remaining tasks after the deletion of a task, as per [this](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=(i.e.%20if%20you%27re%20deleting%20the%20first%20task%20index%201%20from%20the%20list%2C%20the%20index%20of%20the%20next%20task(2)%20should%20set%20to%201)..) requirement.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @carreraprogrammer, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/task.js (hunk 1 above)
Comment (carreraprogrammer) [reply src/modules/task.js:44]: Hi @ShadyShawkat!

The Id is resetting correctly, can you be more specific with the requirement?

https://www.loom.com/share/f8bbd71864f246178f4f04135f78b81e

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @carreraprogrammer , 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Clean/Organized code
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**